### ISSUE TITLE
Undefined Symbols for architecture arm64.

### ISSUE THREAD

**Issue by christ776** [KhronosGroup/MoltenVK#216]
Hi,
I'm trying to compile the latest version in Release mode, but I'm getting the following errors. I'm using Xcode 10 Beta 5.

```Undefined symbols for architecture arm64:
  "mvk::SPIRVToMSLConverter::convert(mvk::SPIRVToMSLConverterContext&, bool, bool, bool)", referenced from:
      MVKShaderModule::convert(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverter::setSPIRV(unsigned int const*, unsigned long)", referenced from:
      MVKShaderModule::MVKShaderModule(MVKDevice*, VkShaderModuleCreateInfo const*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverterContext::alignWith(mvk::SPIRVToMSLConverterContext const&)", referenced from:
      MVKShaderLibraryCache::getShaderLibrary(mvk::SPIRVToMSLConverterContext*, MVKShaderModule*, bool*) in MoltenVK(MoltenVK-arm64-master.o)
      MVKShaderLibraryCache::findShaderLibrary(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-arm64-master.o)
      MVKShaderLibraryCache::merge(MVKShaderLibraryCache*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverterContext::isVertexBufferUsed(unsigned int) const", referenced from:
      MVKGraphicsPipeline::getMTLRenderPipelineDescriptor(VkGraphicsPipelineCreateInfo const*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverterContext::isVertexAttributeLocationUsed(unsigned int) const", referenced from:
      MVKGraphicsPipeline::getMTLRenderPipelineDescriptor(VkGraphicsPipelineCreateInfo const*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverterContext::matches(mvk::SPIRVToMSLConverterContext const&) const", referenced from:
      MVKShaderLibraryCache::getShaderLibrary(mvk::SPIRVToMSLConverterContext*, MVKShaderModule*, bool*) in MoltenVK(MoltenVK-arm64-master.o)
      MVKShaderLibraryCache::findShaderLibrary(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-arm64-master.o)
      MVKShaderLibraryCache::merge(MVKShaderLibraryCache*) in MoltenVK(MoltenVK-arm64-master.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **```

**Comment by billhollings** [comment]
@christ776 

Strange. 

It looks like the `MoltenVKSPIRVToMSLConverter.framework` is either not being built in the Xcode Target Dependency Build Phase...or is not being included in the linking.

Are you using the LunarG SDK...or just MoltenVK repo?

And do you have the same issue on non-beta Xcode versions?

**Comment by Arthapz** [comment]
After disabling iOS build, same error on x86_64 arch

```
Undefined symbols for architecture x86_64:
  "mvk::SPIRVToMSLConverter::convert(mvk::SPIRVToMSLConverterContext&, bool, bool, bool)", referenced from:
      MVKShaderModule::convert(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-x86_64-master.o)
  "mvk::SPIRVToMSLConverter::setSPIRV(unsigned int const*, unsigned long)", referenced from:
      MVKShaderModule::MVKShaderModule(MVKDevice*, VkShaderModuleCreateInfo const*) in MoltenVK(MoltenVK-x86_64-master.o)
  "mvk::SPIRVToMSLConverterContext::alignWith(mvk::SPIRVToMSLConverterContext const&)", referenced from:
      MVKShaderLibraryCache::findShaderLibrary(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-x86_64-master.o)
  "mvk::SPIRVToMSLConverterContext::isVertexBufferUsed(unsigned int) const", referenced from:
      MVKGraphicsPipeline::getMTLRenderPipelineDescriptor(VkGraphicsPipelineCreateInfo const*) in MoltenVK(MoltenVK-x86_64-master.o)
  "mvk::SPIRVToMSLConverterContext::isVertexAttributeLocationUsed(unsigned int) const", referenced from:
      MVKGraphicsPipeline::getMTLRenderPipelineDescriptor(VkGraphicsPipelineCreateInfo const*) in MoltenVK(MoltenVK-x86_64-master.o)
  "mvk::SPIRVToMSLConverterContext::matches(mvk::SPIRVToMSLConverterContext const&) const", referenced from:
      MVKShaderLibraryCache::findShaderLibrary(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-x86_64-master.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by vitor251093** [comment]
I have the same issue while trying to run the Proton build script in macOS, which builds the official MoltenVK repo. This is how it builds it:

```
./fetchDependencies
xcodebuild -scheme 'MoltenVK (Release)' build
```

**Comment by billhollings** [comment]
@vitor251093 What is your version environment.I'm not able to replicate this using any combination of Xcode 9/10 & macOS 10.13/10.14.

**Comment by BaconWrappedShrimpWithEyes** [comment]
I'm having the same issue compiling Proton on macOS 10.14 with Xcode 10, command line tools for Xcode 9.4.1

`"mvk::SPIRVToMSLConverter::convert(mvk::SPIRVToMSLConverterContext&, bool, bool, bool)", referenced from:
      MVKShaderModule::convert(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverter::setSPIRV(unsigned int const*, unsigned long)", referenced from:
      MVKShaderModule::MVKShaderModule(MVKDevice*, VkShaderModuleCreateInfo const*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverterContext::alignUsageWith(mvk::SPIRVToMSLConverterContext const&)", referenced from:
      MVKShaderLibraryCache::getShaderLibrary(mvk::SPIRVToMSLConverterContext*, MVKShaderModule*, bool*) in MoltenVK(MoltenVK-arm64-master.o)
      MVKShaderLibraryCache::findShaderLibrary(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-arm64-master.o)
      MVKShaderLibraryCache::merge(MVKShaderLibraryCache*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverterContext::isVertexBufferUsed(unsigned int) const", referenced from:
      MVKGraphicsPipeline::getMTLRenderPipelineDescriptor(VkGraphicsPipelineCreateInfo const*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverterContext::isVertexAttributeLocationUsed(unsigned int) const", referenced from:
      MVKGraphicsPipeline::getMTLRenderPipelineDescriptor(VkGraphicsPipelineCreateInfo const*) in MoltenVK(MoltenVK-arm64-master.o)
  "mvk::SPIRVToMSLConverterContext::matches(mvk::SPIRVToMSLConverterContext const&) const", referenced from:
      MVKShaderLibraryCache::getShaderLibrary(mvk::SPIRVToMSLConverterContext*, MVKShaderModule*, bool*) in MoltenVK(MoltenVK-arm64-master.o)
      MVKShaderLibraryCache::findShaderLibrary(mvk::SPIRVToMSLConverterContext*) in MoltenVK(MoltenVK-arm64-master.o)
      MVKShaderLibraryCache::merge(MVKShaderLibraryCache*) in MoltenVK(MoltenVK-arm64-master.o)
ld: symbol(s) not found for architecture arm64

** Build Failed **

The following build commands failed:
	PhaseScriptExecution Create\ Dynamic\ Library /Users/[USERNAME]/Library/Developer/Xcode/DerivedData/MoltenVKPackaging-cpsqswzvkpzutrhdaebfwzmkishz/Build/Intermediates.noindex/MoltenVK.build/Release-iphoneos/MoltenVK-iOS.build/Script-A9731FAD1EDDAE39006B7298.sh
(1 failure)`

**Comment by hagabaka** [comment]
This was fixed in #227, but Proton's MoltenVK submodule is at 8d9ee9b which doesn't have those changes. After I checked out c8a9841 in the submodule the build succeeded.

**Comment by billhollings** [comment]
Looks like the Xcode 10 update in PR #227 has fixed everything in this issue. Closing this now.
